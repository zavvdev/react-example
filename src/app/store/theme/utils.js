import { localStorageService } from "app/services/LocalStorageService";
import { IS_DARK_THEME_LOCAL_STORAGE_KEY } from "app/store/theme/config";

export const saveDarkModeState = (darkModeState) => {
  localStorageService.set(IS_DARK_THEME_LOCAL_STORAGE_KEY, `${darkModeState}`);
};

export const getDarkModeState = () => {
  const rawStorageValue = localStorageService.get(
    IS_DARK_THEME_LOCAL_STORAGE_KEY,
  );
  return !!JSON.parse(rawStorageValue);
};
