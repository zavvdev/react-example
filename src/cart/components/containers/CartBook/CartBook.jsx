import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { useCartBookStyles } from "cart/components/containers/CartBook/CartBook.styles";
import { Shared } from "cart/gateway/input";
import { CART_I18N_NAMESPACE } from "cart/i18n/config";

export function CartBook({ cover, title, author, price, onRemoveFromCart }) {
  const classes = useCartBookStyles();
  const { t } = useTranslation(CART_I18N_NAMESPACE);

  return (
    <div className={classes.root}>
      <div className={classes.inner}>
        <img src={cover} alt={title} className={classes.cover} />
        <div>
          <Shared.Typography className={classes.title} tag="h2">
            {title}
          </Shared.Typography>
          <div>
            <Shared.Typography tag="i">{author}</Shared.Typography>
            <Shared.Typography className={classes.price} tag="div">
              {price}
            </Shared.Typography>
          </div>
        </div>
      </div>
      <button
        type="button"
        onClick={onRemoveFromCart}
        className={classes.removeBtn}
      >
        {t("cartBook.remove")}
      </button>
    </div>
  );
}

CartBook.propTypes = {
  cover: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  onRemoveFromCart: PropTypes.func.isRequired,
};
