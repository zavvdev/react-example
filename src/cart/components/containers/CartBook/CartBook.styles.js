import { createUseStyles } from "react-jss";

export const useCartBookStyles = createUseStyles((theme) => ({
  root: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    borderBottom: `1px solid ${theme.borderColor}`,
    marginBottom: "16px",
    paddingBottom: "16px",
  },
  inner: {
    display: "flex",
    alignItems: "center",
    "@media screen and (max-width: 450px)": {
      display: "block",
    },
  },
  cover: {
    height: "90px",
    marginRight: "16px",
  },
  title: {
    fontSize: "15px",
    margin: "0 0 5px 0",
  },
  removeBtn: {
    border: "none",
    background: "none",
    cursor: "pointer",
    height: "40px",
    minWidth: "50px",
  },
  price: {
    fontWeight: "bold",
    marginTop: "8px",
  },
}));
