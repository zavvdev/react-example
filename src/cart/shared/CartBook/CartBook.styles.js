import { createUseStyles } from "react-jss";

export const useCartBookStyles = createUseStyles((theme) => ({
  root: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    borderBottom: `1px solid ${theme.palette.gray.dark}`,
    marginBottom: "16px",
    paddingBottom: "16px",
  },
  inner: {
    display: "flex",
    alignItems: "center",
    "@media screen and (max-width: 500px)": {
      display: "block",
    },
  },
  cover: {
    height: "90px",
    marginRight: "16px",
    "@media screen and (max-width: 500px)": {
      marginBottom: "8px",
    },
  },
  title: {
    fontSize: "15px",
    margin: "0 0 5px 0",
  },
  removeBtn: {
    border: "none",
    background: "none",
    cursor: "pointer",
    height: "40px",
    minWidth: "50px",
  },
  price: {
    fontWeight: "bold",
    marginTop: "8px",
  },
}));
