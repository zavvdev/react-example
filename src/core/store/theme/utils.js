import { LOCAL_STORAGE_KEYS } from "core/config/localStorage";
import { localStorageService } from "core/services/LocalStorageService";

export const saveDarkModeState = (darkModeState) => {
  localStorageService.set(
    LOCAL_STORAGE_KEYS.isThemeDarkMode,
    `${darkModeState}`,
  );
};

export const getDarkModeState = () => {
  const rawStorageValue = localStorageService.get(
    LOCAL_STORAGE_KEYS.isThemeDarkMode,
  );
  return !!JSON.parse(rawStorageValue);
};
