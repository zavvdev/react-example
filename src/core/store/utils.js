import { all, call, spawn } from "redux-saga/effects";
import { errorTrackingService } from "core/services/ErrorTrackingService";

export const combineMiddleware = (middlewareList) => function* combiner() {
  const getRootGenerator = (middleware) => function* rootGenerator() {
    while (true) {
      try {
        yield call(middleware);
        break;
      } catch (e) {
        errorTrackingService.reportError(e);
      }
    }
  };

  yield all(middlewareList.map((middleware) => {
    return spawn(getRootGenerator(middleware));
  }));
};
