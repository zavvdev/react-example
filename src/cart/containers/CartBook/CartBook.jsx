import PropTypes from "prop-types";
import { useCartBookStyles } from "cart/containers/CartBook/CartBook.styles";
import { Shared } from "cart/gateway/input";
import { useCartTranslation } from "cart/i18n/useCartTranslation";

export function CartBook({ cover, title, author, price, onRemoveFromCart }) {
  const classes = useCartBookStyles();
  const { t } = useCartTranslation();

  return (
    <div className={classes.root}>
      <div className={classes.inner}>
        <img src={cover} alt={title} className={classes.cover} />
        <div>
          <Shared.Typography className={classes.title} tag="h2">
            {title}
          </Shared.Typography>
          <div>
            <Shared.Typography tag="i">{author}</Shared.Typography>
            <Shared.Typography className={classes.price} tag="div">
              {price}
            </Shared.Typography>
          </div>
        </div>
      </div>
      <button
        type="button"
        onClick={onRemoveFromCart}
        className={classes.removeBtn}
      >
        {t("cartBook.remove")}
      </button>
    </div>
  );
}

CartBook.propTypes = {
  cover: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  onRemoveFromCart: PropTypes.func.isRequired,
};
