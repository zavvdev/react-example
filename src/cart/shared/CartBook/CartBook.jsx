import PropTypes from "prop-types";
import { useCartBookStyles } from "cart/shared/CartBook/CartBook.styles";
import { useCartTranslation } from "cart/i18n/useCartTranslation";

export function CartBook({ cover, title, author, price, onRemoveFromCart }) {
  const classes = useCartBookStyles();
  const { t } = useCartTranslation();

  return (
    <div className={classes.root}>
      <div className={classes.inner}>
        <img src={cover} alt={title} className={classes.cover} />
        <div>
          <h2 className={classes.title}>{title}</h2>
          <div>
            <i>{author}</i>
            <div className={classes.price}>{price}</div>
          </div>
        </div>
      </div>
      <button
        type="button"
        onClick={onRemoveFromCart}
        className={classes.removeBtn}
      >
        {t("cartBook.remove")}
      </button>
    </div>
  );
}

CartBook.propTypes = {
  cover: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  onRemoveFromCart: PropTypes.func.isRequired,
};
