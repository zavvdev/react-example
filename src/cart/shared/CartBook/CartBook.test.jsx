import { fireEvent, screen } from "@testing-library/react";
import { CartBook } from "cart/shared/CartBook/CartBook";
import { testUtils } from "__tests__/utils";

const properties = {
  cover: "https://mock.com/42",
  title: "Mock title",
  author: "Mock Author",
  price: "$42",
  onRemoveFromCart: jest.fn(),
};

const component = (
  <CartBook
    cover={properties.cover}
    title={properties.title}
    author={properties.author}
    price={properties.price}
    onRemoveFromCart={properties.onRemoveFromCart}
  />
);

test("should render CartBook", () => {
  const result = testUtils.render(component);
  expect(result.asFragment()).toMatchSnapshot();
});

test("should have title, price & author", () => {
  testUtils.render(component);
  expect(screen.getByText(properties.title)).toBeInTheDocument();
  expect(screen.getByText(properties.price)).toBeInTheDocument();
  expect(screen.getByText(properties.author)).toBeInTheDocument();
});

test("should call onRemoveFromCart function", () => {
  testUtils.render(component);
  const removeButton = screen.getByText("cartBook.remove");
  fireEvent.click(removeButton);
  expect(properties.onRemoveFromCart).toBeCalled();
});
