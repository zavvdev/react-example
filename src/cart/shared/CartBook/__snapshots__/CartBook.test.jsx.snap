// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartBook 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="inner"
    >
      <img
        alt="Mock title"
        class="cover"
        src="https://mock.com/42"
      />
      <div>
        <h2
          class="title"
        >
          Mock title
        </h2>
        <div>
          <i>
            Mock Author
          </i>
          <div
            class="price"
          >
            $42
          </div>
        </div>
      </div>
    </div>
    <button
      class="removeBtn"
      type="button"
    >
      cartBook.remove
    </button>
  </div>
</DocumentFragment>
`;
